### PR TITLE
make deps.zig compatible with gyro

### PR DESCRIPTION
**Pull request by mattnite**
This patch adds an `addAllTo()` function under the `pkgs` namespace, this is so that package maintainers can more easily use both gyro and zigmod. It only adds zig packages to a `LibExeObjStep`, so it is functionally equivalent to gyro as well.

looks like zig fmt picked up some other fixes along the way.

### CODE REVIEW

File: src/cmd/fetch.zig [hunk 1]
```diff
@@ -370,7 +370,15 @@ fn print_pkgs(alloc: std.mem.Allocator, w: std.fs.File.Writer, m: zigmod.Module)
...
         try w.print("    pub const {s} = package_data._{s};\n", .{ ident, d.id[0..12] });
     }
-    try w.writeAll("}");
+
+    try w.writeAll("    pub fn addAllTo(artifact: *std.build.LibExeObjStep) void {\n");
+    if (m.deps.len == 0)
+        try w.writeAll("        _ = artifact;\n");
+
+    for (m.deps) |d|
+        try w.print("        artifact.addPackage(package_data._{s});\n", .{d.id[0..12]});
```
Comment (nektro) [inline src/cmd/fetch.zig:379]: this should be `package_data._{s}.pkg.?`

File: src/cmd/fetch.zig [hunk 2]
```diff
@@ -370,7 +370,15 @@ fn print_pkgs(alloc: std.mem.Allocator, w: std.fs.File.Writer, m: zigmod.Module)
         const ident = try zig_name_from_pkg_name(alloc, d.name);
         try w.print("    pub const {s} = package_data._{s};\n", .{ ident, d.id[0..12] });
     }
-    try w.writeAll("}");
+
+    try w.writeAll("    pub fn addAllTo(artifact: *std.build.LibExeObjStep) void {\n");
```
Comment (nektro) [inline src/cmd/fetch.zig:374]: please add a spacer `\n` so the function definition isnt scrunched up against the decl list

Comment (nektro) [reply src/cmd/fetch.zig:374]: feel free to add a comment too that the method is for gyro compat and doesnt call the full `deps.addAllTo(artifact);`

**Comment by nektro** [comment]
why not have users do `if (@hasDecl(deps, "addAllTo") deps.addAllTo(exe) else deps.pkgs.addAllTo(exe);` or polyfill gyro to move `addAllTo` up into the root namespace given that if the purpose of this was to avoid the if statement, users should prefer the more featureful function?

**Comment by mattnite** [comment]
my thinking was that:
- `deps.addAllTo`: add all dependencies
- `deps.pkgs.addAllTo`: add all _packages_

and so with a reduced scope you get reduced functionality since zigmod's root `addAllTo` does more than just add packages.

**Comment by nektro** [comment]
right and why would you only want to add packages when dependencies depend on the other things also being added?

**Comment by mattnite** [comment]
It’s for those who only depend on pure zig packages, they can explicitly reduce the scope. It’s better this way because the same function with the same scope has the same behaviour, where if I were to polyfill gyro, the implementation would be the different because it doesn’t deal with C libraries the same way.

Users who aren’t aware of the differences could end up with link errors if I polyfill gyro, and that’s far more unsettling than a compiler error if you’re not used to this stuff. What I could do is generate a root declaration of `addAllTo` that results in a nice compiler error that gives context.

**Pull request closed without merging**